src/messages.c

```diff
@@ -0,0 +1,54 @@
+// NOTE: This file has been generated automatically by “update-message-index.py”.
+//       Do not edit manually!
+
+#include <stdlib.h>
+#include "messages.h"
+
+static const MessageEntry xkb_messages[] = {
+    {107, "unrecognized-keysym", "Warn on unrecognized keysyms."},
+    {183, "cannot-infer-key-type", "Warn if no key type can be inferred."},
+    {286, "undefined-key-type", "Warn if using an undefined key type."},
+    {305, "non-base-group-name", "Warn if a group name was defined for group other than the first one."},
+    {461, "conflicting-key-symbol", "Warn if there are conflicting keysyms while merging keys."},
+    {489, "numeric-keysym", "Warn on numeric keysym (other than 0-9)."},
+    {516, "extra-symbols-ignored", "TODO: add description"},
+    {700, "multiple-groups-at-once", "Warn if a key defines multiple groups at once."},
+    {769, "invalid-syntax", "The syntax is invalid and the file cannot be parsed."},
+    {770, "undefined-keycode", "TODO: add description"},
+    {800, "conflicting-modmap", "Warn if there are conflicting modmap definitions."},
+    {883, "conflicting-key-action", "Warn if there are conflicting actions while merging keys."},
+    {893, "conflicting-key-type", "Warn if there are conflicting key types while merging groups."},
+    {935, "conflicting-key-fields", "Warn if there are conflicting fields while merging keys."},
+    {965, "unresolved-keymap-symbol", "Warn if using a symbol not defined in the keymap."},
+    {XKB_INVALID_MESSAGE_INDEX, NULL, NULL}
+};
+
+xkb_message_code_t
+XkbMessageCode(enum xkb_message_index message_index)
+{
+    if (message_index < 0 || message_index >= XKB_LOG_MESSAGE_MAX_INDEX) {
+        return XKB_INVALID_MESSAGE_INDEX;
+    } else {
+        return xkb_messages[message_index].code;
+    }
+}
+
+const char *
+XkbMessageIdentifier(enum xkb_message_index message_index)
+{
+    if (message_index < 0 || message_index >= XKB_LOG_MESSAGE_MAX_INDEX) {
+        return NULL;
+    } else {
+        return xkb_messages[message_index].identifier;
+    }
+}
+
+const char *
+XkbMessageDescription(enum xkb_message_index message_index)
+{
+    if (message_index < 0 || message_index >= XKB_LOG_MESSAGE_MAX_INDEX) {
+        return NULL;
+    } else {
+        return xkb_messages[message_index].description;
+    }
+}
```

Comment:
the `typedefs` and `XkbCamelCase` style naming is a leftover from the xkbcomp sources imported, iirc all of modern libxkbcommon uses `struct foo` and `xkb_foo`. Let's keep that here please.

Reply:
Sure, let’s be consistent